src/taipy/core/submission/submission.py

```diff
@@ -130,56 +145,46 @@ def __gt__(self, other):
     def __ge__(self, other):
         return self.creation_date.timestamp() >= other.creation_date.timestamp()
 
-    def _update_submission_status(self, _: Job):
-        abandoned = False
-        canceled = False
-        blocked = False
-        pending = False
-        running = False
-        completed = False
-
-        for job in self.jobs:
-            if not job:
-                continue
-            if job.is_failed():
-                self.submission_status = SubmissionStatus.FAILED  # type: ignore
-                return
-            if job.is_canceled():
-                canceled = True
-                continue
-            if job.is_blocked():
-                blocked = True
-                continue
-            if job.is_pending() or job.is_submitted():
-                pending = True
-                continue
-            if job.is_running():
-                running = True
-                continue
-            if job.is_completed() or job.is_skipped():
-                completed = True
-                continue
-            if job.is_abandoned():
-                abandoned = True
-        if canceled:
+    def _update_submission_status(self, job: Job):
```

Comment:
Good idea, but the current implementation indicates that we're passing the job to the callbacks needed to be run and we don't have an approach to say "we will be updating the submission status now" to determine that we should pass the status only instead of the whole job